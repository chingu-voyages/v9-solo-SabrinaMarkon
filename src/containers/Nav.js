import React from 'react';
import { NavLink } from 'react-router-dom';

export default function Nav() {

  return (
      <nav className="navbar navbar-inverse navbar-fixed-top">
        <div className="container">
          <div className="navbar-header">
            <button type="button" className="navbar-toggle">
              <span className="icon-bar"></span>
              <span className="icon-bar"></span>
              <span className="icon-bar"></span>                        
            </button>
          </div> 
            <div className="single-page-nav sticky-wrapper" id="tmNavbar">
              <ul className="nav navbar-nav">
                <li key="nav-0" className="nav__item">
                  <NavLink className="nav__link" exact activeClassName="active" to="/">Home</NavLink>
                </li>
                <li key="nav-1" className="nav__item">
                  <NavLink className="nav__link" exact activeClassName="active" to="/howitworks">How It Works</NavLink>
                </li>
                <li key="nav-2" className="nav__item">
                  <NavLink className="nav__link" exact activeClassName="active" to="/signup">Signup</NavLink>
                </li>
                <li key="nav-3" className="nav__item">
                  <NavLink className="nav__link" exact activeClassName="active" to="/login">Login</NavLink>
                </li>
                <li key="nav-4" className="nav__item">
                  <NavLink className="nav__link" activeClassName="active" to="/postjob">Post a Job!</NavLink>
                </li>
                <li key="nav-5" className="nav__item">
                  <a href="#section4">Contact</a>
                </li>
              </ul>
            </div>
          </div>
      </nav>
  )
}