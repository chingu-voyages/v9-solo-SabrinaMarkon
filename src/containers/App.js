import React, { Component, ErrorBoundary } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import HowItWorks from '../components/HowItWorks';
import Signup from '../components/Signup';
import Login from '../components/Login';
import PostJob from '../components/PostJob';
import NotFound from '../components/NotFound';

import Nav from '../containers/Nav';
import Header from '../components/Header';
import Footer from '../components/Footer';

export default class App extends Component {

  render() {
    return (
        <BrowserRouter>
          <div className="container">
            <Nav />
            <Header />
            <Switch>
              <Route exact path="/" />
              <Route exact path="/howitworks" component={HowItWorks} />
              <Route exact path="/signup" component={Signup} />
              <Route exact path="/login" component={Login} />
              <Route exact path="/postjob" component={PostJob} />
              <Route component={NotFound} />
            </Switch>
            <Footer />
          </div>
        </BrowserRouter>
    )
  }
}




