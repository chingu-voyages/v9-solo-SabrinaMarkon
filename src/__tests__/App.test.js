import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

// components
import App from '../containers/App';
import Nav from '../containers/Nav';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});

describe('Testing main App component rendering', () => {
  it('should render a React Fragment in the App component.', () => {
    const fragment = document.createElement('Fragment');
    ReactDOM.render(<App />, fragment);
    ReactDOM.unmountComponentAtNode(fragment);
  });

  it('should render 1 child Nav component', () => {
    // one way to check that a component is rendered:containsMatchingElement
    expect(wrapper.containsMatchingElement(<Nav />)).toEqual(true);
    // expect(wrapper.find('Nav')).toHaveLength(1);
  });

  it('should render 1 child Header component', () => {
    expect(wrapper.find('Header')).toHaveLength(1);
  }); 

  it('should render 1 BrowserRouter element', () => {
    expect(wrapper.find('BrowserRouter')).toHaveLength(1);
  });
  
  it('should render 6 Route elements', () => {
    expect(wrapper.find('Route')).toHaveLength(6);
  });

  it('should render 1 child Footer component', () => {
    expect(wrapper.find('Footer')).toHaveLength(1);
  });

  it('it should match the App snapshot', () => {
    // If snapshot exists, compare. If not, create the snapshot.
    expect(wrapper).toMatchSnapshot();
  });

});