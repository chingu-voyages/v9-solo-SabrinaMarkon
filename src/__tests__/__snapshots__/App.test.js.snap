// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing main App component rendering it should match the App snapshot 1`] = `ShallowWrapper {}`;
